chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/UnattestedCriteriaWithDataReviewer.java

```diff
@@ -20,75 +22,58 @@ public void review(CertifiedProductSearchDetails listing) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Additional Software"));
+                    cert.getAdditionalSoftware().clear();
                 }
                 if (cert.getTestDataUsed() != null && cert.getTestDataUsed().size() > 0) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Test Data"));
+                    cert.getTestDataUsed().clear();
                 }
                 if (cert.getTestFunctionality() != null && cert.getTestFunctionality().size() > 0) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Test Functionality"));
+                    cert.getTestFunctionality().clear();
                 }
                 if (cert.getTestProcedures() != null && cert.getTestProcedures().size() > 0) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Test Procedures"));
+                    cert.getTestProcedures().clear();
                 }
                 if (cert.getTestStandards() != null && cert.getTestStandards().size() > 0) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Test Standards"));
+                    cert.getTestStandards().clear();
                 }
                 if (cert.getTestToolsUsed() != null && cert.getTestToolsUsed().size() > 0) {
-                    listing.getErrorMessages()
-                    .add(msgUtil.getMessage("listing.criteria.falseCriteriaHasData",
-                            cert.getNumber(), "Test Tools"));
+                    cert.getTestToolsUsed().clear();
                 }
 
                 if (listing.getSed() != null && listing.getSed().getTestTasks() != null
                         && listing.getSed().getTestTasks().size() > 0) {
                     for (TestTask tt : listing.getSed().getTestTasks()) {
+                        ArrayList<CertificationCriterion> remove = new ArrayList<CertificationCriterion>();
                         for (CertificationCriterion ttCriteria : tt.getCriteria()) {
                             if (ttCriteria.getNumber() != null && ttCriteria.getNumber().equals(cert.getNumber())) {
-                                listing.getErrorMessages().add(msgUtil.getMessage(
-                                        "listing.criteria.falseCriteriaHasData",
-                                        cert.getNumber(), "Test Tasks"));
+                                remove.add(ttCriteria);
```

Comment:
I think this is the correct logic for this, can someone confirm?

Reply:
I think that looks okay. Did you test it in the real world?